ConcurrencyHomework/src/main/multiThread/ThreadedFileReader.java

```diff
@@ -0,0 +1,71 @@
+package main.multiThread;
+
+import main.entities.Gender;
+import main.entities.Person;
+
+import java.io.*;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+
+/**
+ * Threaded implementation for the homework requirement
+ */
+
+public class ThreadedFileReader implements Runnable {
+    String inputPath;
+    String outputPath;
+
+    public ThreadedFileReader(String inputPath, String outputPath) {
+        this.inputPath = inputPath;
+        this.outputPath = outputPath;
+    }
+
+    /**
+     * This method reads the data from the file received as an argument and creates new Person objects.
+     * Then, if one person's birthday is today, he/she will be written in an output file containing the happy birthday message,
+     * the name and the year that person was born
+     *
+     */
+    @Override
+    public void run() {
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(String.valueOf(Paths.get(inputPath)))));
+             BufferedWriter bw = new BufferedWriter(new OutputStreamWriter((new FileOutputStream(String.valueOf(Paths.get(outputPath))))))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] splitLine = line.split(",");
```

Comment:
Toata logica de parsare si intrepretare a obiectelor rezultate ar fi trebuit scoasa in o alta metoda, eventual in alta clasa. Astfel, codul devenea mai testabil si evitai duplicarea de cod(ai scrisa aceeasi logica atat in `ThreadedFileReader` cat si `FileReader`.